src/iosrv/init.c

```diff
@@ -207,7 +213,12 @@ dss_tgt_nr_get(int ncores, int nr)
 	 * dss_tgt_offload_xs_nr offload XS. Calculate the nr_default
 	 * as the number of main XS based on number of cores.
 	 */
-	nr_default = (ncores - dss_sys_xs_nr) / DSS_XS_NR_PER_TGT;
+	if (dss_numa_node == -1)
+		nr_default = (ncores - dss_sys_xs_nr) / DSS_XS_NR_PER_TGT;
+	else
+		nr_default = (((ncores - dss_sys_xs_nr) - dss_core_offset) /
```

Comment:
if dss_numa_node is set, then seems better to ingore the dss_core_offset. for example with 2 numa node each with 16 cores, then config with dss_core_offset of 16 will cause it get incorrect result.

looks better to juse use orignal code for either case.

Reply:
I preserved the dss_core_offset because it looked to me like we would still want to preserve that capability even when binding to cores within a NUMA node. Agreed that if we don't want to be able to specify an offset, that can be removed in a future refinement.